### PR TITLE
Added new Purdue Computer Science Faculty!

### PR DESCRIPTION
**Pull request by emkin11**
Added new Purdue CS Faculty:
Raymond A. Yeh
Ruqi Zhange
Ryan Newton
Eric Samperton
Brian Bullins
Mohammadkazem Taram

Additionally, there are other senior Purdue CS faculty who are listed in csrankings .csv file on github **but don't show up in CSrankings** (they have valid DBLP and have publications in venues listed by CSRankings). Here are the faculty names:

Mikhail Atallah
Bharat Bhargava
Z. Berkay Celik
Douglas Comer
Susanne Hambrusch
Majid Kazemian
Anuran Makur
Vernon Rego
Elisha Sacks
Eugene Spafford
Rajiv Khanna

# Contributing to CSrankings

Thanks for contributing to CSrankings! Please read and indicate you agree with **all** these guidelines to getting your pull request accepted. Note that pull requests may take some time to get merged (up to three months, as processing has moved to a quarterly cadence).

**If you find CSrankings useful, [please consider becoming a sponsor](https://github.com/sponsors/csrankings).**

**NOTE:** IF YOU DO NOT FOLLOW THE STEPS BELOW, YOUR COMMIT MAY BE SUMMARILY REJECTED.

1. _Use a reasonable title that explains what the PR corresponds to (as in, not "Update csrankings-x.csv")._

1. _Do not modify any files except `csrankings-[a-z].csv` or (if needed) `country-info.csv`._

1. _Do not use Excel to edit any .csv files; Excel incorrectly tries to
convert some Google Scholar entries to formulas, corrupting the
database. Use a text editor like emacs or NotePad instead._

1. _Insert new faculty **in alphabetical order** (not at the end) in the appropriate `csrankings-[a-z].csv` files._ **Do not modify `csrankings.csv`, which is auto-generated.**

1. _Check to make sure that you have no spaces after commas, or any missing fields._

1. _Check to make sure the home page is correct._

1. _Make sure the Google Scholar IDs are just the alphanumeric identifier (not a URL or with `&hl=en`)._

1. _Check to make sure the name corresponds to the DBLP entry (look it up at http://dblp.org)._

1. _If a faculty member is not in a CS department or similar, include a comment explaining how they meet the inclusion criteria (see below)._

1. _Read and check **all** the boxes below by filling them in with an X._

**Inclusion criteria**

- [x] Make sure that any faculty you add meet the inclusion
criteria. Eligible faculty include only full-time, tenure-track research
faculty members on a given campus who can *solely* advise PhD students in
Computer Science. Faculty not in a CS department or similar who can
advise PhD students in CS can be included regardless of their home
department. Faculty should also have a 75%+ time appointment (check
`old/industry.csv` for faculty who are now more than 25% in industry).

**Updating an affiliation or home page**

- [x] Update affiliations, home pages, and Google Scholar entries by modifying `csrankings-[a-z].csv`. For the Google Scholar entry, just use the alphanumeric identifier in the middle of the URL. If none is there, put `NOSCHOLARPAGE`.

**Adding one or more faculty members (including an entire department)**

- [ ] If the department is not yet listed in CSrankings, the entire CS faculty needs to be added (not just one faculty member).

- [x] Enter each faculty member's [DBLP](http://dblp.org) name, home page, and Google Scholar entry (just the alphanumeric identifier, not the whole URL) by modifying `csrankings-[a-z].csv` (**the letters correspond to the first letter of the faculty members' names**); include disambiguation suffixes like `0001` as needed. If the faculty entry is currently ambiguous, please do not include them. Send mail to the DBLP maintainers (dblp@dagstuhl.de) with a few publications by a particular faculty member; also, open an issue so that when the DBLP database is updated, that faculty member's information can be added.

- [ ] If DBLP has multiple entries for this person, *all of them need to be listed*. Do not update `dblp-aliases.csv`.

- [ ] If the institution you are adding is not in the US,
update `country-info.csv` and add *all* of the faculty in the CS department.

**(Advanced) Quick contribution via a shallow clone** 

We recommend that you use the GitHub web user interface to make changes. However, it may be more convenient
to clone the repository for larger-scale changes.

However, a full clone of the CSrankings repository is almost 2GB.  To
contribute a change without creating a full local clone of the
CSrankings repo, you can perform a _shallow clone_.  To do so, follow these
steps:

1. Fork the CSrankings repo.  If you have an existing fork, but it is not up to date with the main repository, this technique may not work.  If necessary, delete and re-create your fork to get it up to date.  (Do **not** delete your existing fork if it has unmerged changes you want to preserve!)

1. Do a shallow clone of your fork: `git clone --depth 1 https://github.com/yourusername/CSrankings`.  This will only download the most recent commit, not the full git history.

1. Make your changes on a branch, push them to your clone, and create a pull request on GitHub as usual.

If you want to make another contribution and some time has passed, perform these steps again, creating a fresh fork and shallow clone.

### CODE REVIEW

**Comment by emeryberger** [comment]
With respect to senior faculty in the database - have you changed the start year? For example, Gene Spafford's most recent CSrankings publication is from 2005. (I set the start year to 2005; see below.) Please set the maximum range and try again.

Screen Shot 2022-09-11 at 6 29 41 PM

**Comment by emkin11** [comment]
Hello and thanks for your explanation and patience.

I believe this corrects all entries except:

Z. Berkay Celik

He is listed as Berkay Celik in the .csv file.
Homepage: https://www.cs.purdue.edu/people/faculty/zcelik.html
Dblp: https://dblp.org/pid/143/2298.html
Google scholar: g1I269gAAAAJ

Thanks again,

Emily

From: Emery Berger ***@***.***>
Date: Sunday, September 11, 2022 at 6:31 PM
To: emeryberger/CSrankings ***@***.***>
Cc: Emily A Kinsell ***@***.***>, Author ***@***.***>
Subject: Re: [emeryberger/CSrankings] Added new Purdue Computer Science Faculty! (PR #5020)
---- External Email: Use caution with attachments, links, or sharing data ----


With respect to senior faculty in the database - have you changed the start year? For example, Gene Spafford's most recent CSrankings publication is from 2005. (I set the start year to 2005; see below.) Please set the maximum range and try again.

[Image removed by sender. Screen Shot 2022-09-11 at 6 29 41 PM]<https://user-images.githubusercontent.com/1612723/189551581-31778e61-b623-4873-88d1-7f440700357b.png>

**Comment by emeryberger** [comment]
Fixed that entry with https://github.com/emeryberger/CSrankings/commit/9aa4bfad6e3b476ba54db1c2a02db9fa5771ef1e.

File: csrankings-r.csv [hunk 1]
```diff
@@ -1383,6 +1385,7 @@ Ryan Kok Leong Ko,University of Queensland,https://researchers.uq.edu.au/researc
 Ryan Layer,University of Colorado Boulder,http://layerlab.org,AXnmFTIAAAAJ
 Ryan M. Layer,University of Colorado Boulder,http://layerlab.org,AXnmFTIAAAAJ
 Ryan Marcus,University of Pennsylvania,https://rmarcus.info/,vPOl-IwAAAAJ
+Ryan Newton, Purdue University, https://www.cs.purdue.edu/people/faculty/rrnewton.html, Bp6dm-oAAAAJ
```
Comment (emeryberger) [inline csrankings-r.csv]: All spaces need to be removed after commas (in every entry). Per LinkedIn, Ryan is not a TT faculty member at Purdue (listed as Adjunct) - please clarify.

**Review by emeryberger — CHANGES_REQUESTED** [review]
Please fix space issues (no spaces after commas) and address clarification question.

File: csrankings-r.csv (hunk 1 above)
Comment (emeryberger) [reply csrankings-r.csv]: Please address this issue ASAP; I am closing old PRs by the end of this week.

Comment (emkin11) [reply csrankings-r.csv]: Ryan Newton is an TT Associate Professor in the Department of Computer Science and in the School of Electrical and Computer Engineering at Purdue University. 
https://www.cs.purdue.edu/people/faculty/rrnewton.html
https://engineering.purdue.edu/ECE/People/ptProfile?resource_id=247461

Spaces have been removed.

File: csrankings-r.csv [hunk 2]
```diff
@@ -1385,7 +1384,7 @@ Ryan Kok Leong Ko,University of Queensland,https://researchers.uq.edu.au/researc
 Ryan Layer,University of Colorado Boulder,http://layerlab.org,AXnmFTIAAAAJ
 Ryan M. Layer,University of Colorado Boulder,http://layerlab.org,AXnmFTIAAAAJ
 Ryan Marcus,University of Pennsylvania,https://rmarcus.info/,vPOl-IwAAAAJ
-Ryan Newton, Purdue University, https://www.cs.purdue.edu/people/faculty/rrnewton.html, Bp6dm-oAAAAJ
+Ryan Newton,Purdue University,https://www.cs.purdue.edu/people/faculty/rrnewton.html,Bp6dm-oAAAAJ
```
Comment (emeryberger) [inline csrankings-r.csv]: Ryan's LinkedIn does not indicate he is full-time at Purdue. Please remove.
https://www.linkedin.com/in/rrnewton/

**Review by emeryberger — CHANGES_REQUESTED** [review]

**Review by emkin11 — COMMENTED** [review]
made changes requested

**Pull request merged**